libs/marinade-client-rs/src/transactions/transaction_executors.rs

```diff
@@ -207,14 +204,13 @@ pub fn execute_transaction_builder(
     preflight_config: RpcSendTransactionConfig,
     blockhash_commitment: CommitmentLevel,
     simulate: bool,
-    print_only: bool,
+    print: bool,
     blockhash_failure_retries: Option<u16>,
 ) -> anyhow::Result<()> {
-    warn_text_simulate_print_only(simulate, print_only);
+    warn_text_simulate_print(simulate, print);
 
-    if print_only {
+    if print {
         print_base64(&transaction_builder.instructions())?;
-        return Ok(());
```

Comment:
My considering is following:
When `print` then most probably instead of keypair will be a pubkey. That means execution cannot succeed and there is no much reason to process it - i.e., it's the behaviour at `execute_anchor_builders_with_config`.
When `print` then it could be reasonable to try to simulate to validate if the printed transaction has some chance to be executed on chain (sometimes later).
When there is no print then either execution or simulation is executed (and signature verification makes sense).

I'm still unsure if this is correct.
When `print` is true and `simulate` is true it's simulated and printed.
When `print` is true and `simulate` is false it's executed.

Reply:
I understand that with `print` set and `simulate` not set it might not be the most predictable behavior, but that's the downside of allowing the user to choose any combination and rely on them to do it reasonably. To make it a bit more predictable, `print-only` was renamed to `print` so it wouldn't make a false impression, I was even considering something like `also-print`/`print-tx-log` or something similar to make it even more explicit that there's no `-only`.

If we wanted to play it very safe (and you're worried about making a mistake when using the CLI) we could've gone for the other option where setting either of the flags (or the one of them which won't be removed) results in not executing anything.

Do you think I should rework it to some safer option or are we fine with the fact that the flags are simply independent of each other and you can freely `print` or not as you choose, regardless of whether you simulate or execute? Or do you maybe see some (reasonably complex) solution which wasn't mentioned yet?

Reply:
I think we spent pretty long time on this and so just let's put it out of the table :-)

My consideration is usage.
The most usually when one wants to print he needs to put the output to spl gov or other multisig system.
Default behaviour is now with `--print` to execute. Which most probably fails (as providing most probably the pubkey and not keypair). With printing the execution makes no sense to me.

I feel better default behaviour should be either `only print` or add `simulate` to printing as an additional step.